crates/sui-adapter/src/programmable_transactions/execution.rs

```diff
@@ -436,6 +447,92 @@ fn execute_move_publish<E: fmt::Debug, S: StorageView<E>, Mode: ExecutionMode>(
     Ok(values)
 }
 
+/// Upgrade a Move package.  Returns an `UpgradeReceipt` for the upgraded package on success.
+fn execute_move_upgrade<E: fmt::Debug, S: StorageView<E>>(
+    context: &mut ExecutionContext<E, S>,
+    upgrade_ticket_arg: Argument,
+    dep_ids: Vec<ObjectID>,
+    module_bytes: Vec<Vec<u8>>,
+) -> Result<Value, ExecutionError> {
+    // Check that package upgrades are supported.
+    context
+        .protocol_config
+        .check_package_upgrades_supported()
+        .map_err(|_| {
+            ExecutionError::from_kind(ExecutionErrorKind::PackageUpgradeError(
```

Comment:
I think we probably just want a general error for "this feature isn't supported". Might be a bad idea to start carving it out per feature.